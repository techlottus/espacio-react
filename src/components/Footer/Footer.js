

import React, { useEffect } from 'react'

export const Footer = React.memo(({data, onLink}) => {

  const footerRef = React.createRef();

  useEffect(() => {
    footerRef.current.data = {
      urlLogo: data.urlLogo || '',
      text: data.text || '',
      privacy: data.privacy || '',
      terms: data.terms || '',
    };
    footerRef.current.addEventListener('onLink', onLink)
  }, [data]) // eslint-disable-line react-hooks/exhaustive-deps

  return (
    <>
      <lottus-footer ref={footerRef}></lottus-footer>
    </>
  )
})

