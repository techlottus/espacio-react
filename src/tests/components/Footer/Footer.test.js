import React from "react";
import { mount } from "enzyme";
import { Footer } from "../../../components/Footer/Footer";
import { Provider } from "react-redux";
import "@testing-library/jest-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

let initState = {
  data: {
    urlLogo: null,
    text: null,
    privacy: null,
    terms: null,
  },
};

let store = mockStore(initState);
const handleMount = (initState) => {
  store = mockStore(initState);
  return mount(
    <Provider store={store}>
      <Footer data={initState.data} />
    </Provider>
  );
};

describe("Test <Footer />", () => {
  test("should init component", () => {
    const wrapper = handleMount(initState);
    expect(wrapper).toMatchSnapshot();
  });
});
