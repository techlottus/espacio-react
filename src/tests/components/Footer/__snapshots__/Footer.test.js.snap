// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Footer /> should init component 1`] = `<lottus-footer />`;
